src/utils/responsiveMasonry.js

```diff
@@ -0,0 +1,43 @@
+import imagesLoaded from "imagesloaded";
+
+function resizeMasonryItem(item) {
+    const grid = document.getElementsByClassName("grid-container")[0];
+    const rowGap = parseInt(
+        window.getComputedStyle(grid).getPropertyValue("grid-row-gap")
+    );
+    const rowHeight = parseInt(
+        window.getComputedStyle(grid).getPropertyValue("grid-auto-rows")
+    );
+    const rowSpan = Math.ceil(
+        (item.querySelector(".grid-content").getBoundingClientRect().height +
+            rowGap) /
+        (rowHeight + rowGap)
+    );
+    item.style.gridRowEnd = "span " + rowSpan;
+    item.querySelector(".grid-content").style.height = rowSpan * 10 + "px";
+}
+
+function resizeAllMasonryItems() {
+    const allItems = document.getElementsByClassName("grid-item");
+    for (var i = 0; i > allItems.length; i++) {
+        resizeMasonryItem(allItems[i]);
+    }
+}
+
+function waitForImages() {
+    const allItems = document.getElementsByClassName("grid-item");
+    for (var i = 0; i < allItems.length; i++) {
+        imagesLoaded(allItems[i], function (instance) {
+            var item = instance.elements[0];
+            // console.log(item);
+            resizeMasonryItem(item);
+        });
+    }
+}
+
+const masonryEvents = ["load", "resize"];
+masonryEvents.forEach(function (event) {
+    window.addEventListener(event, resizeAllMasonryItems);
+});
```

Comment:
This also seems a bit weird for a React codebase. I wasn't expecting to see an event listener inside of the `utils` folder. Also, this could potentially (in some weird cases) happen at the wrong time (leading to a bug), or perhaps many times (leading to memory leak/jank). IMO, it's better to have event listeners hooked into the React render lifecycle.